tests/ledctl/ledctl_cmd.py

```diff
@@ -0,0 +1,130 @@
+# SPDX-License-Identifier: GPL-3.0-or-later
+# Copyright (C) 2023 Intel Corporation.
+
+import subprocess
+import logging
+import re
+
+LOGGER = logging.getLogger(__name__)
+
+class Slot:
+    def __init__(self, cntrl_type, slot_id, state, device_node):
+        self.cntrl_type = cntrl_type
+        self.slot = slot_id
+        self.state = state.lower()
+        self.device_node = None
+        if device_node.startswith('/dev/'):
+            self.device_node = device_node
+
+    def __str__(self):
+        return "slot: %s state: %s device: %s" % (self.slot, self.state, self.device_node)
+
+class LedctlCmd:
+
+    slot_mgmt_ctrls = ["SCSI", "VMD", "NPEM"]
+
+    def __init__(self, ledctl_bin=["none"], slot_filters="none", controller_filters="none"):
+        # We cares about first entry (not full valdation but it is enough for test purposes)
+        self.bin = ledctl_bin[0]
+
+        # The purpose or slot filters is exclude unsupported but recognized slots, apply it
+        # globally.
+        self.slot_filters = slot_filters
+
+        # Give possibility to skip controllers.
+        self.slot_ctrls = [i for i in self.slot_mgmt_ctrls if i not in controller_filters]
+
+    def run_ledctl_cmd(self, params: list, output=False, check=True):
+        params.insert(0, "sudo")
+        params.insert(1, self.bin)
+
+        LOGGER.debug(f"Command: {params}")
+        return subprocess.run(params, capture_output=output, check=check)
+
+    def run_ledctl_cmd_decode(self, params: list):
+        result = self.run_ledctl_cmd(params, output=True)
+        out = result.stdout.decode("utf-8")
+
+        LOGGER.debug(f"Command returned:\n {out}")
+        return out
+
+    # Ledctl Commands
+
+    def set_slot_state(self, slot: Slot, state):
+        self.run_ledctl_cmd([ "--set-slot", "--controller-type", slot.cntrl_type,
+                              "--slot", slot.slot,  "--state", state])
+
+    def get_slot(self, slot: Slot, slot_filter):
```

Comment:
what about slot_filter param? It is not used

Reply:
good one!